Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,447 @@
+    Assert::AreEqual(weather_descriptor::none, metar.weather_group.at(0).descriptor);
+    Assert::AreEqual(size_t(1), metar.weather_group.at(0).phenomena.size());
+    Assert::AreEqual(weather_phenomena::rain, metar.weather_group.at(0).phenomena.at(0));
+    metar.weather_group.clear();
+
+    // -TSRA
+    std::string m2("-TSRA ");
+    metar::parsers::parse_weather(metar, m2);
+    Assert::AreEqual(weather_intensity::light, metar.weather_group.at(0).intensity);
+    Assert::AreEqual(weather_descriptor::thunderstorm, metar.weather_group.at(0).descriptor);
+    Assert::AreEqual(size_t(1), metar.weather_group.at(0).phenomena.size());
+    Assert::AreEqual(weather_phenomena::rain, metar.weather_group.at(0).phenomena.at(0));
+    metar.weather_group.clear();
+
+    // +SN
+    std::string m3("+SN ");
+    metar::parsers::parse_weather(metar, m3);
+    Assert::AreEqual(weather_intensity::heavy, metar.weather_group.at(0).intensity);
+    Assert::AreEqual(weather_descriptor::none, metar.weather_group.at(0).descriptor);
+    Assert::AreEqual(size_t(1), metar.weather_group.at(0).phenomena.size());
+    Assert::AreEqual(weather_phenomena::snow, metar.weather_group.at(0).phenomena.at(0));
+    metar.weather_group.clear();
+
+    // VCFG
+    std::string m4("VCFG ");
+    metar::parsers::parse_weather(metar, m4);
+    Assert::AreEqual(weather_intensity::in_the_vicinity, metar.weather_group.at(0).intensity);
+    Assert::AreEqual(weather_descriptor::none, metar.weather_group.at(0).descriptor);
+    Assert::AreEqual(size_t(1), metar.weather_group.at(0).phenomena.size());
+    Assert::AreEqual(weather_phenomena::fog, metar.weather_group.at(0).phenomena.at(0));
+    metar.weather_group.clear();
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_SkyCondition()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(size_t(0), metar.sky_condition_group.size());
+
+    // CLR
+    std::string m1("CLR ");
+    metar::parsers::parse_sky_condition(metar, m1);
+    Assert::AreEqual(sky_cover_type::clear_below_12000, metar.sky_condition_group.at(0).sky_cover);
+    Assert::AreEqual(sky_cover_cloud_type::none, metar.sky_condition_group.at(0).cloud_type);
+    Assert::AreEqual(UINT32_MAX, metar.sky_condition_group.at(0).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(0).unit);
+    metar.sky_condition_group.clear();
+
+    // SKC
+    std::string m2("SKC ");
+    metar::parsers::parse_sky_condition(metar, m2);
+    Assert::AreEqual(sky_cover_type::sky_clear, metar.sky_condition_group.at(0).sky_cover);
+    Assert::AreEqual(sky_cover_cloud_type::none, metar.sky_condition_group.at(0).cloud_type);
+    Assert::AreEqual(UINT32_MAX, metar.sky_condition_group.at(0).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(0).unit);
+    metar.sky_condition_group.clear();
+
+    // VV030
+    std::string m3("VV030 ");
```

Comment:
Vertical visibility is the visibility into an indefinite ceiling; I'm not sure a vertical visibility of 3000 ft makes sense.  Can we change this to a more realistic 300 ft?